Add reporting support to ContainerGebSpec

matrei opened the pull request:
Introduces a `boolean reportingEnabled = false` property in `ContainerGebSpec` that subclasses can override to enable reporting functionality.

To enable reporting, you must also specify the `reportsDir` property. This can typically be configured in `resources/GebConfig.groovy` or via the `geb.reportsDir` system property.

matrei committed:
feat: Add reporting support to ContainerGebSpec

Introduces a `boolean reportingEnabled = false` property in `ContainerGebSpec` that subclasses can override to enable reporting functionality.

To enable reporting, you must also specify the `reportsDir` property. This can typically be configured in `resources/GebConfig.groovy` or via the `geb.reportsDir` system property.

matrei committed:
refactor: Use `isReportingEnabled()` for better readability

sbglasius commented on src/testFixtures/groovy/grails/plugin/geb/GebTestManagerProvider.groovy:
```diff
@@ -0,0 +1,36 @@
+        }
+        return REPORTING_INSTANCE
+    }
+}
```
I believe this can be reduced to:

```
@CompileStatic
class GebTestManagerProvider {

    @Lazy
    static volatile GebTestManager instance = { 
        new SpockGebTestManagerBuilder().build() 
    }()
        
    @Lazy
    static volatile GebTestManager reportingInstance = {
        new SpockGebTestManagerBuilder()
                .withReportingEnabled(true)
                .build() 
    }()

    private GebTestManagerProvider() {}
    
}
```
That makes it a bit more Groovy 😉

matrei replied:
Excellent suggestion, the groovier the better!

sbglasius reviewed (approved):
You could consider my replacement for `GebTestManagerProvider` or leave it as is. I think it looks good! ⭐

matrei committed:
refactor: PR feedback, use `@Lazy` annotation

Refactored `GebTestManagerProvider` to leverage
Groovy's `@Lazy` annotation for the `GebTestManager` fields.

This change simplifies the code and results in the use of
the initialization-on-demand holder pattern instead of
using synchronization with double-checked locking pattern
for the field initialization.

sbglasius reviewed (approved):
🤩
